### PR TITLE
Support authentification

### PR DESCRIPTION
**Pull request by PiR-B**

### CODE REVIEW

**Comment by TuningYourCode** [comment]
hi, 
is creation of docker repository without authentication still possible after your change? E.g. we have instances which don't require authentication.
I can have a look but will take some time :) Worst case we have to introduce a small switch-case in the code to keep the old behavior by default.

**Comment by PiR-B** [comment]
I made some changes to my PR to better suit your needs. I'm not even sure why I did that, as I don't need it either!

By the way, do you know if the API documentation is up to date?
For the POST method, it doesn't mention the 'writePolicy' parameter. However, when I check the GET method, it does mention it.
After conducting some tests, I found that this parameter is necessary to ensure idempotency

**Comment by TuningYourCode** [comment]
well, i wouldn't trust the api documentation 100%. In major parts it should be fine but i can remember that we had to implement somewhere a workaround.
Not sure if it was related to nexus repository core or if it was a (community) plugin. We will also raise the nexus version with our next release to the latest one - so if in doubt look what happens with the latest nexus version that it is idempotent :)

File: manifests/resource/repository/docker/proxy.pp [hunk 1]
```diff
@@ -54,6 +55,13 @@
   Enum['present', 'absent'] $ensure = 'present',
   Boolean $http_client_blocked = false,
   Boolean $http_client_auto_block = true,
+  Optional[Struct[{
+        type => Enum['username', 'ntlm '],
```
Comment (TuningYourCode) [inline manifests/resource/repository/docker/proxy.pp:59]: `type => Enum['username', 'ntlm'],` i guess the space is not supposed to be there - else looks good for me now :)

Comment (PiR-B) [reply manifests/resource/repository/docker/proxy.pp]: oupsy !

**Pull request merged**